macrosynergy/signal/target_positions.py

```diff
@@ -123,14 +239,17 @@ def target_positions(df: pd.DataFrame, cids: List[str], xcat_sig: str, ctypes: L
     :param <List[str]> ctypes: contract types that are traded across markets. They should
         correspond to return categories in the dataframe if the `ret` argument is
         appended. Examples are 'FX' or 'EQ'.
-    :param <Dict[Any, Dict]> baskets: dictionary of basket dictionaries. The inner
-        dictionary takes a string of form <cross_section>_<contract_type> as key and a
-        list of string of the same form as value. The key labels the basket. The value
-        defines the contracts that are used for forming the basket. Pe default the
-        contract have equal weights. An example would be:
-        {{'APC_FX' : ['AUD_FX', ''NZD_FX', 'JPY_FX']},
-         {'APC_EQ' : ['AUD_EQ', ''CNY_EQ', 'INR_EQ', 'JPY_EQ']}}
-        # Todo: has yet to be implemented
+    :param <dict> baskets: a dictionary containing the name of each basket and the
+        corresponding constituents. The key is of the form
+        <cross_section>_<contract_type> and the value will be a list of the associated
+        contracts. The key labels the basket. The value defines the contracts that are
+        used for forming the basket. The default weighting method is for equal weights.
+        An example would be:
+        {'APC_FX' : ['AUD_FX', 'NZD_FX', 'JPY_FX'],
+         'APC_EQ' : ['AUD_EQ', 'CNY_EQ', 'INR_EQ', 'JPY_EQ']}
+    :param: <List[pd.DataFrame] or pd.DataFrame> df_c_wgts: weight dataframe used to
```

Comment:
There should be no need for passing any additional data frames, since the weights produced by the Basket class are already in the standard format and can be pass to the function via the argument df.

Reply:
Change made.